### ISSUE TITLE
if_delegate_has_method is deprecated and has been removed from scikit-learn

### ISSUE THREAD

**Issue by penguinpee** [manuel-calzolari/sklearn-genetic#37]
Recently, scikit-learn has been updated to version 1.3.0 in Fedora Rawhide (development branch). Now tests for sklearn-genetic fail:

```
==================================== ERRORS ====================================
__________ ERROR collecting genetic_selection/tests/test_selection.py __________
ImportError while importing test module '/builddir/build/BUILD/sklearn-genetic-0.5.1/genetic_selection/tests/test_selection.py'.
Hint: make sure your test modules/packages have valid Python names.
Traceback:
/usr/lib64/python3.12/importlib/__init__.py:90: in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
genetic_selection/__init__.py:16: in <module>
    from .gscv import GeneticSelectionCV
genetic_selection/gscv.py:22: in <module>
    from sklearn.utils.metaestimators import if_delegate_has_method
E   ImportError: cannot import name 'if_delegate_has_method' from 'sklearn.utils.metaestimators' (/usr/lib64/python3.12/site-packages/sklearn/utils/metaestimators.py)
=========================== short test summary info ============================
ERROR genetic_selection/tests/test_selection.py
!!!!!!!!!!!!!!!!!!!! Interrupted: 1 error during collection !!!!!!!!!!!!!!!!!!!!
=============================== 1 error in 0.64s ===============================
```

`if_delegate_has_method` has been [deprecated](https://github.com/scikit-learn/scikit-learn/blob/9aaed498795f68e5956ea762fef9c440ca9eb239/sklearn/utils/metaestimators.py#L122) since version 1.1 and is now (1.3.0) [removed](https://github.com/scikit-learn/scikit-learn/commit/9260f510abcc9574f2383fc01e02ca7e677d6cb7) from `metaestimators.py`.

I tried a quick patch, but didn't succeed. I might look into it again, time permitting.

**Comment by detrin** [comment]
Yeah same, I will try to fix this.

**Comment by detrin** [comment]
I fixed the error. I will make PR, would be easier to get into the contributors group

**Comment by musicinmybrain** [comment]
> I fixed the error. I will make PR, would be easier to get into the contributors group

Would you mind posting the PR? We’d like to fix the [Fedora Linux package](https://src.fedoraproject.org/rpms/python-sklearn-genetic), and a working PR would be useful as a downstream patch even before it is merged here. Thanks!

**Comment by manuel-calzolari** [comment]
Sorry for the delay but I am quite busy at this time. I will release an updated version of the library next week to fix compatibility issues like this.

**Comment by musicinmybrain** [comment]
> Sorry for the delay but I am quite busy at this time. I will release an updated version of the library next week to fix compatibility issues like this.

Thanks, that sounds great!

**Comment by detrin** [comment]
> > I fixed the error. I will make PR, would be easier to get into the contributors group
> 
> Would you mind posting the PR? We’d like to fix the [Fedora Linux package](https://src.fedoraproject.org/rpms/python-sklearn-genetic), and a working PR would be useful as a downstream patch even before it is merged here. Thanks!

yeah, I will make it. I think the simplest fix is to get rid of the decorator and make the check of the method inside wrapped rmethod

**Comment by detrin** [comment]
will be solved in https://github.com/manuel-calzolari/sklearn-genetic/pull/41

please @manuel-calzolari have a look at my PR

**Comment by manuel-calzolari** [comment]
Closed with commit 12ee9b2. Thank you all!
